modules/javafx.web/src/main/java/com/sun/javafx/webkit/prism/RTImage.java

```diff
@@ -47,10 +49,13 @@
 final class RTImage extends PrismImage implements ResourceFactoryListener {
     private RTTexture txt;
     private final int width, height;
-    private boolean listenerAdded = false;
+    private WeakReference<ResourceFactory> listenerAdded = null;
```

Comment:
may be rename the variable to `resourceFactoryRef`

Reply:
I renamed it here, and in `WCPageBackBufferImpl`, to `registeredWithFactory` which is the same name as is used in `PrismMediaFrameHandler`.